frame/contracts/src/exec.rs

```diff
@@ -2114,7 +2111,12 @@ mod tests {
 			let forbidden_call = Call::Balances(BalanceCall::transfer { dest: CHARLIE, value: 22 });
 
 			// simple cases: direct call
-			assert_err!(ctx.ext.call_runtime(forbidden_call.clone()), BadOrigin);
+			assert_err!(
+				ctx.ext.call_runtime(forbidden_call.clone()),
+				sp_runtime::DispatchErrorWithPostInfo::from(
+					frame_system::Error::<Test>::CallFiltered
+				)
+			);
```

Comment:
Why the explicit `from` call? The macro should call `into` just as for the old code?

```suggestion
			assert_err!(
				ctx.ext.call_runtime(forbidden_call.clone()),
				frame_system::<Error<Test>>::CallFiltered
			);
```

Reply:
damn I was always writing `.into()` making the compiler failing... thanks

Reply:
hmm that should work, too. Let's see what the CI says

Reply:
no what I mean is that I tried to write:
```diff
diff --git a/frame/contracts/src/exec.rs b/frame/contracts/src/exec.rs
index 7ef1aec2df..8d6ea52561 100644
--- a/frame/contracts/src/exec.rs
+++ b/frame/contracts/src/exec.rs
@@ -2113,7 +2113,7 @@ mod tests {
                        // simple cases: direct call
                        assert_err!(
                                ctx.ext.call_runtime(forbidden_call.clone()),
-                               frame_system::Error::<Test>::CallFiltered
+                               frame_system::Error::<Test>::CallFiltered.into()
                        );
 
                        // as part of a patch: return is OK (but it interrupted the batch)
```

And it failed with
```
error[E0282]: type annotations needed
    --> frame/contracts/src/exec.rs:2116:47
     |
2116 |                 frame_system::Error::<Test>::CallFiltered.into()
     |                 ------------------------------------------^^^^--
     |                 |                                         |
     |                 |                                         cannot infer type for type parameter `T` declared on the trait `Into`
     |                 this method call resolves to `T`
     |
     = note: type must be known at this point

```

I didn't realize they were calling `into` inside the macro

Reply:
Ahh okay. Make sense. It is only called on the right hand argument, though.